gtfs-realtime/proto/gtfs-realtime.proto

```diff
@@ -1033,3 +1030,99 @@ message Shape {
+		optional int32 propagated_modification_delay = 3 [default = 0];
+
+    // A list of replacement stops, replacing those of the original trip. The length of the new stop times may be less, the same, or greater than the number of replaced stop times. 
+		repeated ReplacementStop replacement_stops = 4;
+
+		// The extensions namespace allows 3rd-party developers to extend the
+		// GTFS Realtime Specification in order to add and evaluate new features and
+		// modifications to the spec.
+		extensions 1000 to 1999;
+
+		// The following extension IDs are reserved for private use by any organization.
+		extensions 9000 to 9999;
+	}
+
+  // An arbitrary string to identify this detour. The modifications_id MUST be unique for all currently active detours. The ID of each replacement trip is generated by concatenating modifications_id + '_' + trip_id.
+  required string modifications_id = 1;
+
+   // A list of trips affected by this detour. All trips linked by those trip_ids MUST have the same stop pattern. Two trips have the same stop pattern, if they visit the same stops in the same order, and have identical stop sequences.
+  repeated string trip_ids = 2;
+
+  // Dates on which the detour occurs, in the YYYYMMDD format. Producers SHOULD only transmit detours occurring within the next week.
+  repeated string service_dates = 3;
+
+  // The ID of the new shape for the modified trips. May refer to a new shape added using a GTFS-RT Shape message, or to an existing shape defined in the GTFS-Static feed’s shapes.txt.
+  optional string shape_id = 4;
+
+  // A list of modifications to apply to the affected trips. 
+  repeated Modification modifications = 5; 
+
+  // The extensions namespace allows 3rd-party developers to extend the
+  // GTFS Realtime Specification in order to add and evaluate new features and
+  // modifications to the spec.
+  extensions 1000 to 1999;
+
+  // The following extension IDs are reserved for private use by any organization.
+  extensions 9000 to 9999;
+}
+
+message ReplacementStop {
+  // The difference in seconds between the arrival time at this stop and the arrival time at the reference stop. The reference stop is the stop prior to start_stop_sequence. If the modification begins at the first stop of the trip, then the first stop of the trip is the reference stop.
+  // This value MUST be monotonically increasing and may only be a negative number if the first stop of the original trip is the reference stop.
+  optional int32 travel_time_to_stop = 1;
+
+  // The replacement stop ID which will now be visited by the trip. May refer to a new stop added using a GTFS-RT Stop message, or to an existing stop defined in the GTFS-Static feed’s stops.txt. The stop MUST have location_type=0 (routable stops).
+  optional string stop_id = 2;
```

Comment:
In the modified trip, we create new `stop_sequence` for that trip. See [here](https://github.com/google/transit/pull/403/files/9a0d6d5e9c790c2f22a88809a2993a9c2c14077d#diff-abc780f2e1f0fd946abf169b0bb5d5ed06c33c695a028c7119043c0584eb7759R17). 

Having a separate `stop_sequence` in `ReplacementStop` could also be valid. Do you think it would be better? One big downside of this is that if the `stop_sequence` in the static GTFS is increasing by one at each stop_time it would mean you can't have a higher number of replacement stops than stops are cancelled.

Reply:
Yes, but if a `stop_id` appears twice in `ReplacementStop`, how do we know which `travel_time_to_stop` maps to which `StopTimeUpdate`? I'll try to put together an example.

Reply:
<img width="858" alt="Screenshot 2023-10-23 at 2 47 31 PM" src="https://github.com/google/transit/assets/214921/2226bd5c-f3ad-435b-ad06-8a223a3eae41">


Given that stop F is used twice during the detour, how should we expect this to be represented?

Reply:
In this example in the Modification, the `replacement_stops` is an ordered list so the order is defined there. Exactly like you did in the `replacement_stops` list on the right of the image. 

To target the F stop within a TripUpdate, you would have to use a `stop_sequence` of 3 or 6 in the `StopTimeUpdate`.